src/Kiota.Builder/Writers/CSharp/CodeMethodWriter.cs

```diff
@@ -84,6 +87,17 @@ public override void WriteCodeElement(CodeMethod codeElement, LanguageWriter wri
                                             .ThenBy(x => x.Name)) {
                 writer.WriteLine($"{propWithDefault.Name.ToFirstCharacterUpperCase()} = {propWithDefault.DefaultValue};");
             }
+            if(currentMethod.IsOfKind(CodeMethodKind.Constructor)) {
```

Comment:
The code here seems impacted by the code in https://github.com/microsoft/kiota/pull/322/files#diff-61c9c35ebb5b7e20af529c42cc88f8849f5c4df4e585595b8e7964cee57b343dR336. Consider adding a test to make sure a change in KiotaBuilder is paired with a change here. Or consider a comment in both places to help indicate that the code here is related to the code in KiotaBuilder. 

This comment is applicable to all /writers/**/CodeMethodWriter.cs code files.

Reply:
When we [have a look at the design overview](https://microsoft.github.io/kiota/designoverview) it's clear that writers and refiners, which are downstream of the builder, will be impacted by changes made in the builder. The only "hole" I can see in the docs is it doesn't callout the components names specifically.

Again, the builder's responsibility is to paint the most accurate picture of "here is all the things we should represent in the outcome", refiners apply a "lens" on it to taint it according to the target language. The writers may choose or not to implement specific details. While I'm making all the work on a single front for the 3 prime languages so we can make rapid progress and get early feedback, I'm not expecting the writers to implement every detail from day one, or at all depending on the language's philosophy. I'm not expecting a lot of changes to happen in the builder either after the ground work is done, the action is going to happen in the refiners/writers/libraries for each language.

All that being said, I recognize the code method writers are becoming "hairy", I've contemplated for a while splitting those and introducing new code elements to make distinctions and maintenance easier. Things like "CodeConstructor" and "CodeAccessor". This should lead to dividing up the number of lines in the method writers by 2/3 but might lead to a little duplication as well. Do you have any thoughts on that after going through multiple reviews?